docs/src/references/mqtt-topic-guidelines.md

```diff
@@ -0,0 +1,165 @@
+   * All data from all services excluding those from thin-edge device and child devices
+   * All data from a specific child device excluding everything else
+   * All data from a specific service excluding everything else
+1. The topic structure should be ACL friendly so that rules can be applied
+   to limit child devices and services to access only the data meant for them.
+1. Enable easier extension of topics with further topic suffixes in future:
+   E.g: Support `type` in the topic for measurements like `tedge/measurements/{measurement-type}`
+
+### Nice-to have
+
+1. Dynamic creation/registration of child devices on receipt of the very first data that they send.
+   This is desired at least for immediate child devices, if not for further nested child devices.
+1. Easy to create static routing rules so that it is easy to map a local MQTT topic for a nested child device
+   to the equivalent cloud topic for its twin.
+1. All the existing topics like `tedge/measurements`, `tedge/events` imply that the data received on these
+   must be forwarded to the cloud as well.
+   Currently there is no way to tell thin-edge to just route some data internally and not forward those to the cloud.
+   Since filtering and aggregation on the edge is a very common use-case, especially for local analytics, this is highly desired.
+1. It would be ideal if the context/source of data (tedge device, service or child device)
+   can be understood from the topic directly.
+   For e.g: a topic scheme like `tedge/main/{id}`, `tedge/service/{id}` and `tedge/child/{id}` is more user-friendly
+   than a simpler context agnostic scheme like `tedge/{id}` where `id` can be for any "thing".
+1. Allowing the "things" in a local domain to use their user-friendly unique IDs (within the tedge namespace)
+   in MQTT topics over their globally unique cloud IDs (typically very long and cumbersome) would be desired.
+   Ideally, all translation between user-provided-ids and cloud-twin-ids should be done internally by thin-edge,
+   as long as it doesn't adversely affect any data mapping cost at scale.
+
+### Out of scope
+
+1. Routing different kinds of data to different clouds, e.g: all telemetry to azure and all commands from/to Cumulocity.
+   Even though this requirement is realistic, thin-edge MQTT topics must not be corrupted with cloud specific semantics,
+   and the same requirement will be handled with some external routing mechanism(e.g: routing via bridge topics)
+1. Services on child devices. Service support can be limited to thin-edge devices only.
+
+## Proposals
+
+### Dedicated topics for tedge device, services and child devices
+
+The topics for the thin-edge device, the services running on it and child devices have different prefixes:
+
+For parent: tedge/main/
+For services: tedge/service/<service-id>
+For immediate child devices: tedge/child/<child-id>
+For nested child devices: tedge/child/<child-id>
```

Comment:
Yeah it was supposed to be `tedge/descendent/<child-id>` which was corrected in the next commit.

Reply:
Having a `descendent` seems quite ad-hoc and confusing. What is the motivation to put them so clearly apart?

Reply:
Initially I also wanted to treat all child devices the same(which is why I first wrote `tedge/child` here, but changed it in the following commit). The distinction was made primarily to easily support auto-registration, so that we can support auto-registration for all `tedge/child` messages but not for `tedge/descendent` messages.